### ISSUE TITLE
[Bug Report][3.7.3] VDataTable with custom headers does not align checkbox

### ISSUE THREAD

**Issue by SuperDJ** [vuetifyjs/vuetify#20672]
### Environment
**Vuetify Version:** 3.7.3
**Last working version:** 2.7.2
**Vue Version:** 3.5.12
**Browsers:** Edge 130.0.0.0
**OS:** Windows 10

### Steps to reproduce
When creating a datatable with custom headers, one of which is an aligned checkbox the header will align but not the checkbox. This does not work for both the v-checkbox and v-checkbox-btn

### Expected Behavior
It will align the checkbox when according to the header alignment

### Actual Behavior
The checkbox remains at the start instead of at center or end

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqtVMGO2jAQ/ZVRLuxKkMBS2DZiUXvocbeVOHQlwsEkQ3ExdmRPKBHKv3cSCCUSIC27OUSemTee5zf2THees3HwLU39TYZe6I0I16kShONIA4w2nUSQ6JCYKywdAKFkhHuKvNhoZxS6yDsEligStGVouiNJCkNovYg1ttqwwpwNXRpFG47R79tYHaPIRubkpsQLJX9r9sWoCW2rmO2LVJyYVc0RNh2nDFWU/GM+E9hB6YIi8g4p1VHiJcarudl25qQ5dW0SVAxuZkceWD6I0Sofj4JmUl0/aIrEqP8qsasRZ9PFVqZUYXGbGkuQ4EJkimC337DMDuHuHp7GcHfwAdQChzCtXVBn7L9SUFbppxL5hARJo2HQap8i1kJnCxFTZtEycmJ03gTkKDjw0H3oNtxOVIVbve5zE3/UKQSyGZ7EipP1OZavLCJM0Ep08HqV5bOMrXFmQW+iOvQHl7kuhHJvIfsi+eLpxMDkr6R4eZVujT3Ptvd4lu3nLx8m7Gn7P93Y/l7/fPt7w4/t/w/Nz/v21l+gObhyS2/u/C8p39H27vDCe+rdIOjssCruq0VRjpjjUPHanpJ65fw/zmie4NWheDw7nszl6PCWRKkLgyBONGN45smN9TVSoNN1wFOf5CL/2vcf/X6QSEe1q6PE3PnVNrNIc02vaOtMqeo3+wfz8cW8)

**Comment by MatthewAry** [comment]
VCheckbox has the `display: flex` style applied in a css class which affects alignment. You can work around this by applying the `d-inline-flex` class to the VCheckbox components. [Here is a modified version of your reproduction.](https://play.vuetifyjs.com/#eNqtVFFv2jAQ/isnv9BKJIF20C1qq+1hj+0m8bBKhAfjXIaHsSPbYUQo/32XQBCRAKmseYh8d9/Z393Z33TLnBXRtzwP1wWymD16XOWKe3xONMDjOki554Hnc4W1AyCWhHBPCRNGO6PQJWwfWCBP0dah6dZLrzCG3itfYa8PSyzJ0LVR9eEQ/b4R6hBFMgon1zWeK/lbk0+g9mh71Wx3SMOJWLUcYR04ZXxDKTzkE4Et1C6oErZPaUoRCxTLudkEc68pdWVSVATuZicMLBVitCpBKO7qetJAaiU1BpnCDQGilkjU7VZ03C5ydeJkOmFl7hssbnJjPaSY8UJ52O42rLNjuLmFp2e42fsA2k7HMG1d0Gbsvrqz1K6fipcTz700Gka9/jFixXWRceELi5aQE6PLLqBEToG7wd2g43a8Obg3HLx08YeGxeBtgUex6mh9iuUbjQAmaCU6eLvI8kUKa5zJ/LuojsPRea4ZV+49ZF8l3UCdGpj8lV4sLtJtsafZDh9Osv385cMaezz+T1eOf3h/evzD8cfO/4emd3796M/QHF24pVdP/peU/zH2wfjMexpe0dDZflXdNouqlpiDqLA+I5FauvCPM5qkvCmKdNqRRNfSwRbe5y6OIpFqwpD4ybUNNfpI56uI5N/LrPx6Hz6E91EqnW9dgeJzFzbbzBJNZ7Kqrwulmt/sH/DJyGI=)
